tutorials/timefreq/wavelets_tutorial.py

```diff
@@ -0,0 +1,191 @@
+# It takes input parameter n_seconds and fs, which specify the simulation time and signal
+# sampling rate, respectively.
+# It also takes components which include 'sim_powerlaw',
+# 'sim_oscillation', and 'freq'.
+# These components must be specified separately.
+#
+# The 'sim_powerlaw' function can be found under neurodsp.aperiodic.py.
+# It simulates a power law time series with a specified exponent. We'll also need
+# the key word arguments to pass to the function filter_signal, which can be found under
+# neurodsp.filt.filter.py.
+#
+# The filter_signal function has input parameters sig, fs, pass_type, f_range,
+# filter_type, n_cycles, n_seconds, remove_edges, butterworth_order,
+# print_transitions, plot_properties, and return_filter.
+# The sig parameter defines the time series to be filtered.
+# The fs parameter is the sampling rate.
+# The pass_type parameter specifies the type of filter to apply. The options include
+# bandpass, bandstop, lowpass, and highpass.
+# The f_range parameter defines the cutoff frequency or frequencies used for the filter,
+# specified as f_lo and f_hi.
+# The filter_type parameter is optional and determines whether to use a FIR or IIR
+# filter (the only IIR filter option is a butterworth filter).
+# The n_cycles parameter is the length of the filter, in number of cycles, at the 'f_lo'
+# frequency if using a FIR filter.
+# The n_seconds parameter is the length of the filter, in seconds, if using a FIR filter.
+# The remove_edges parameter is an optional boolean parameter. If set to True, it replaces
+# samples within half the kernel length to be np.nan (only for FIR filters).
+# The butterworth_order parameter is the order of the butterworth filter, if using
+# an IIR filter.
+#
+# For this tutorial, we will use a lowpass filter FIR filter.
+#
+# The 'sim_oscillation' function can be found under neurodsp.periodic.py.
+# It simulates an oscillation, and takes input parameters n_seconds, fs, freq,
+# cycle, phase, and cycle_params.
+# The cycle is specified as sine, and the phase default is 0.
+# The cycle_params are the parameters for the simulated oscillation cycle,
+# which are already stored.
+#
+# We'll also need the function 'create_times' to create an array of time indices.
+# It takes input parameters n_seconds, fs, and start_val.
+# The n_seconds parameter is the length of the signal.
+# The fs parameter is the sampling frequency.
+# The start_val parameter is the starting value for the time definition, and it
+# has a default value of 0.
+
+###################################################################################################
+
+# Simulation settings
+fs = 500
+n_seconds = 10
+
+# Define simulation components
+components = {'sim_powerlaw': {'exponent' : {-2.0}}, {'f_range' : {None}}, {'**filter_kwargs' : {'sig' : {'sim_combined' : {n_seconds, fs}}, {'pass_type': {'lowpass'}}}}, 'sim_oscillation' : {'freq': 10}})
```

Comment:
This line is way, way too long and it doesn't run. The maintainers of NeuroDSP have chosen to make lines have to be less than 100 characters, usually it's less than 80 in python but it doesn't make a huge difference whether it's 80 or 100, it's really about all the text fitting on an average screen.

I'm going to stop reviewing now because of the syntax error on this line so if you could write a comment in the conversations tab and tag @alexrockhill when you're ready for a review that would be great.

Reply:
I'm kind of struggling with this line. I can't figure out how to add the filter key word arguments, and I don't know how to make the dictionary either. It's hard for me to read using the curly bracket notation, but I know that's how it looks in the other Neurodsp tutorials, so I didn't think I should change it. I thought maybe I should add a line above for the filter_signal to define the key word arguments, like this: `filter_signal(sig=sig, fs=fs, pass_type=lowpass,  f_range=10)
`
But I don't know where to add this line, since sig also relies on what gets defined in the components